pkg/apis/computenode/v1alpha1/computenodeopenstack_types.go

```diff
@@ -24,6 +24,17 @@ type ComputeNodeOpenStackSpec struct {
 	InfraDaemonSets []InfraDaemonSet `json:"infraDaemonSets,omitempty"`
 	// Nodes to delete upon scale down
 	NodesToDelete []NodeToDelete `json:"nodesToDelete,omitempty"`
+        // Image used for drain pod which performs compute removal, this is usually
+        // an image which has the openstackclient and osc-placement packages installed.
+        DrainPodImage string `json:"drainPodImage"`
+        // Namespace OpenStack resources are created, default openstack
+        OpenStackNamespace string `json:"openStackNamespace,omitempty"`
+        // openstackclient configmap which holds information to connect to OpenStack API
+        OpenStackClientConfigMap string `json:"openStackClientConfigMap"`
+        // user secrets used to connect to OpenStack API via openstackclient
+        OpenStackClientAdminSecret string `json:"openStackClientAdminSecret"`
+	// Automatic draining of the node, global switch, which can be overwritten on per Node base using NodeToDelete struct. Default: false
+	Drain bool `json:"drain,omitempty"`
```

Comment:
perhaps we can group Drain and DrainPodImage together, something like:
- Drain:
    enabled: true|false
    image: xxx

Reply:
done